### ISSUE TITLE
(Battle)Mage + Regrowth Staff bug abuse

### ISSUE THREAD

**Issue by Leajian** [00-Evan/shattered-pixel-dungeon#725]
### Requirements
Battlemage with **+3** _Mystical Charge_, **+3** _Empowering Scrolls_ and _Staff of Regrowth_

### The problem
After reading a spell from the Unstable Spellbook, the Staff of Regrowth gets an absurd amount of charges before failing to produce fresh plants and grass, effectively ignoring the "buffer" of the non-empowered version of the wand.

### The abuse
This means I can farm all sorts of seeds, ignoring hunger, because I can heal a lot from sungrass and attacking. What is more, attacking will charge the Unstable Spellbook very fast again and this can continue perpetually, without any limit, because mobs will always spawn. The most important seed is Starflower, which is our main target for the abuse (along with some powerful and useful portions too of course). Given the necessary time and patience, someone can farm enough seeds to brew indefinite amount of Potions of Experience, gaining an early advantage. This of course creates an unfair advantage for challenge runs and makes it a viable strategy.

### An example from a run
|Before using Unstable Spellbook|After using Unstable Spellbook|
:--------:|:-----------:
|![Screenshot_2021-03-21-06-24-41-63_6f3fe77889fbecf862f7061166760abf](https://user-images.githubusercontent.com/1816060/121053196-c6099900-c7c3-11eb-8766-7426017b4d64.jpg)| ![Screenshot_2021-03-21-06-24-29-31_6f3fe77889fbecf862f7061166760abf](https://user-images.githubusercontent.com/1816060/121053212-cb66e380-c7c3-11eb-856e-bd69d8df67ba.jpg) |

**Comment by 00-Evan** [comment]
This is because the level of the regrowth staff is being boosted by 3, so you're seeing the charge limit as if the staff was +9. Once the staff gets close to +10 the charges per level become very generous. This is intended and part of the benefit of upgrading regrowth in the first place

**Comment by boothby** [comment]
I'm playing my second seedmage run, and I agree that it's *ridiculously* overpowered.  Even ignoring all the synergies mentioned, getting a staff of regrowth is a massive boon for a run.  With Wand Preservation, a player can swap the staff out, collect hundreds of seeds eating nothing but dewdrops, and swap back to a combat staff.  If they run low, they can run upstairs, farm more seeds, and get back to business.  In my first run, I scored a bunch of extra scrolls from DM-100s; on this run, I scored a bunch of extra food from Monks.  The only downside is the number of turns spent on the ordeal.

In my current run I have at least 10 of every potion (except experience, I think the rarity there is appropriately balanced), a full complement of upgraded potions to kill the last boss, and I'm about to descend to the King out of sheer boredom.  With this many mind vision & invisibility potions, hell is a walk in the park. I wouldn't call this a bug, but a balance issue that encourages a very tedious exploit.

I think it would be fair to lower the usage cap for the Staff of Regrowth, and to the original point, ignore temporary upgrades when computing the usage cap for the Wand (though, Degraded should still kill the grass :smiling_imp:).
